core/src/main/java/com/sequenceiq/cloudbreak/service/stack/DependentRolesHealthCheckService.java

```diff
@@ -0,0 +1,60 @@
+package com.sequenceiq.cloudbreak.service.stack;
+
+import static java.util.stream.Collectors.toList;
+import static java.util.stream.Collectors.toSet;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import org.springframework.stereotype.Service;
+
+import com.sequenceiq.cloudbreak.api.endpoint.v4.stacks.base.InstanceStatus;
+import com.sequenceiq.cloudbreak.cmtemplate.CmTemplateProcessor;
+import com.sequenceiq.cloudbreak.dto.StackDto;
+import com.sequenceiq.cloudbreak.view.InstanceMetadataView;
+
+@Service
+public class DependentRolesHealthCheckService {
+
+    public static final String UNDEFINED_DEPENDENCY = "UNDEFINED_DEPENDENCY";
+
+    private static final Map<String, List<String>> ROLE_DEPENDENCIES = Map.ofEntries(
+            Map.entry("NODEMANAGER", List.of("RESOURCEMANAGER")));
+
+    public Set<String> getDependentComponents(CmTemplateProcessor processor, String hostGroupScaledUp) {
```

Comment:
`hostGroupScaledUp` - why this variable name? 
This is a service which can be used as others as well. Why should the source usage reflect in the variable name.

The implementation in the last patch, which was not dependent on CmTemplateProcessor was fine.

(OK to leave this also ... but with a proper variable name)

Reply:
Rename method to getDependentComponentsForHostGroup.

Reply:
Add javadoc to method.

Reply:
Done, 
Addressed all of them.